inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcher.php

```diff
@@ -279,4 +281,49 @@ public function get_zones() : array {
 		return [ 'all', 'css_and_js' ];
 	}
 
+	/**
+	 * Compress the HTML data before sending it to the resources fetcher
+	 *
+	 * @since 3.9.2
+	 *
+	 * @param string $data HTML data.
+	 *
+	 * @return string
+	 */
+	public function compress( string $data ): string {
```

Comment:
i think we should bail out if string data is empty

Reply:
Done in https://github.com/wp-media/wp-rocket/pull/3985/commits/7d8f945523f7ac7bd0e84fef1e638f0901b354dd